app/controllers/problem_reports_controller.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+class ProblemReportsController < ApplicationController
+  def index
+    @problem_reports = ProblemReport.order(id: :desc)
+    if params[:all_problems]
+      @filter = 'All'
+    elsif params[:critical_problems]
+      @filter = 'Unfixed and Critical'
+      @problem_reports = filter_by(is_critical: true, is_fixed: false)
+    else
+      @filter = 'Unfixed and Fixable'
+      @problem_reports = filter_by(is_fixable: true, is_fixed: false)
+    end
+  end
+
+  def new; end
+
+  def create
+    if new_report.save
+      redirect_to problem_reports_path,
+                  notice: 'New Problem Report Created'
+    else
+      redirect_to new_problem_report_path,
+                  alert: 'Fail To Make New Problem Report!'
+    end
+  end
+
+  def update
+    @report = ProblemReport.find(params[:id])
+    update_remarks if params[:remarks]
+    update_bool_attr
+    redirect_to problem_reports_path
+  end
+
+  private
+
+  def filter_by(condition)
+    @problem_reports.where(condition)
+  end
+
+  def update_remarks
+    @report.update(last_update_user_id: current_user.id,
+                   remarks: params[:remarks])
+  end
+
+  def boolean_params
+    params.permit(:is_fixable, :is_fixed, :is_blocked, :is_critical)
+          .to_h
+          .map { |k, v| { k => @report[k] ^ v } }
```

Comment:
I suggest the more readable `.map { |k, v| [k, @report[k] ^ v] }.to_h`

Reply:
Oh and I thought `v.nil? ? @report[k] : !@report[k]` has clearer intent than `@report[k] ^ v`